### PR TITLE
Add kube-apiserver, kube-controller-manager, kube-scheduler and etcd to glossary.

### PR DESCRIPTION
**Pull request by ttz21**
Kubecon 2017

Add glossary terms for kube-apiserver, kube-controller-manager, kube-scheduler and etcd using existing text in docs/concepts/overview/components/ page.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6600)

### CODE REVIEW

File: _data/glossary/controller.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+id: controller
+name: controller
```
Comment (tonglil) [inline _data/glossary/controller.yaml]: s/controller/Controller.

File: _data/glossary/controller.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+full-link: /docs/admin/kube-controller-manager/
+related:
+- controller-manager
+tags:
+- architecture
+- fundamental
+short-description: >
+  A control loop that watches the shared state of the cluster through the {% glossary_tooltip text="apiserver" term_id="kube-apiserver" %} and makes changes attempting to move the current state towards the desired state. Examples of controllers that ship with Kubernetes today are the replication controller, endpoints controller, namespace controller, and serviceaccounts controller
```
Comment (tonglil) [inline _data/glossary/controller.yaml]: Please consolidate the examples in the long description.

File: _data/glossary/kube-scheduler.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+id: kube-scheduler
+name: kube-scheduler
+full-link: /docs/reference/generated/kube-scheduler/
+aka:
+- scheduler
+tags:
+- architecture
+short-description: >
+  Component on the master that watches newly created pods that have no node assigned, and selects a node for them to run on.
```
Comment (tonglil) [inline _data/glossary/kube-scheduler.yaml]: A `long-description` is required.

File: _data/glossary/controller.yaml (hunk 2 above)
Comment (tonglil) [reply _data/glossary/controller.yaml]: Without it, [the page](https://deploy-preview-6600--kubernetes-io-master-staging.netlify.com/docs/concepts/overview/components/) will not render properly.

**Comment by zacharysarah** [comment]
@ttz21 👋 Please resolve the merge conflicts so we can proceed.

**Comment by zacharysarah** [comment]
Manually merging due to PR age.

**Pull request merged**